nuto/mechanics/cell/Assembler.h

```diff
@@ -0,0 +1,87 @@
+#pragma once
+#include "nuto/mechanics/dofs/DofContainer.h"
+#include "nuto/mechanics/dofs/DofVector.h"
+#include "nuto/mechanics/dofs/DofMatrix.h"
+#include "nuto/mechanics/dofs/DofMatrixSparse.h"
+
+namespace NuTo
+{
+
+//! Assembles an internal NuTo::DofVector<double> from arbitrary contributions and provides access to it
+class VectorAssembler
+{
+public:
+    //! ctor
+    //! @param sizes number of global dofs for each dof type
+    VectorAssembler(DofContainer<int> sizes = {});
+
+    //! resizes the internal DofVector
+    //! @param sizes number of global dofs for each dof type
+    void Resize(DofContainer<int> sizes);
+
+    //! adds an arbitrary contribution `v` to the internal DofVector
+    //! @param v contribution, e.g. a local element/cell vector
+    //! @param numbering mapping from local (0...numbering.size()) to global (0...sizes)
+    //! @param dofTypes specific dof types to assemble - an empty value will assemble all available dof types from the
+    //! numbering
+    void Add(const DofVector<double>& v, const DofVector<int>& numbering, std::vector<DofType> dofTypes = {});
```

Comment:
Why do we need the dofTypes? I would assume that the DofVector should only include the dof types to be numbered

Reply:
I would not have made this assumption. But it seems like a nice idea and we can certainly do that!